scripts/lint-pr-checks.sh

```
@@ -0,0 +1,246 @@
+    --exclude-dir=".git" \
+    --exclude-dir="DerivedData" \
+    --exclude-dir="node_modules" \
+    --exclude-dir=".build" \
+    --exclude-dir="build" \
+    --exclude-dir="llama.cpp" \
+    --exclude-dir=".venv" \
+    2>/dev/null | grep -v "PROJECT_ROOT" | grep -v "# Or use" | head -10 || true)
+
+if [[ -n "$hardcoded" ]]; then
+    echo -e "   ${YELLOW}WARNING: Found hardcoded user paths:${NC}"
+    echo "$hardcoded" | while read line; do
```

Shellcheck (0.11.0)

[info] read without -r will mangle backslashes.

(SC2162)